### PR TITLE
Add proper sitemaps

### PR DESCRIPTION
**Pull request by kylegach**
- Refactor docs-versions
    - Add `number` and `isPreRelease` fields
    - Export `latestVersion`
    - Improve types & comments
- Update frontpage targets
    - `build`: Now uses built-in executor from `@nx/next`. Old functionality now in `prepare-and-build`.
    - `build-only`: Replaced by `build`
    - `post-build`: Targets and scripts to run after build
    - `prepare-and-build`: Runs `prepare` and `post-build` targets along with `build`
    - `prepare`: Targets and scripts to run before build
    - _These changes were made in advance of the sitemap needs, but done separately to aid explanation_
- Generate sitemaps
    - Replaces existing implementation
    - See below

## Output

Generates (G) or copies (C) the following:
_These are all linked to the deploy preview files_

- /public
  - /sitemap
    - [/sitemap.xml](https://deploy-preview-22--storybook-web.netlify.app/sitemap/sitemap.xml) (G)
    - [/sitemap-0.xml](https://deploy-preview-22--storybook-web.netlify.app/sitemap/sitemap-0.xml) (G)
    - [/sitemap-all.xml](https://deploy-preview-22--storybook-web.netlify.app/sitemap/sitemap-all.xml) (G)
    - [/sitemap-all-0.xml](https://deploy-preview-22--storybook-web.netlify.app/sitemap/sitemap-all-0.xml) (G)
    - [/addons/sitemap.xml](https://deploy-preview-22--storybook-web.netlify.app/sitemap/addons/sitemap.xml) (C, old frontpage)
    - [/blog/sitemap.xml](https://deploy-preview-22--storybook-web.netlify.app/sitemap/blog/sitemap.xml) (C)
    - [/showcase/sitemap.xml](https://deploy-preview-22--storybook-web.netlify.app/sitemap/showcase/sitemap.xml) (C)
    - [/tutorials/sitemap.xml](https://deploy-preview-22--storybook-web.netlify.app/sitemap/tutorials/sitemap.xml) (C)

### CODE REVIEW

File: apps/frontpage/lib/get-sitemap-from-tree.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import { generateDocsTree } from './get-tree';
+import { TreeProps, latestVersion } from '@utils';
+
+const docsTree = generateDocsTree();
+
+const startSitemap: MetadataRoute.Sitemap = [];
+
+function getUrlObjectForSlug(slug: string): MetadataRoute.Sitemap[0] {
+  return {
+    url: `https://storybook.js.org${slug.replace(
```
Comment (kylegach) [inline apps/frontpage/lib/get-sitemap-from-tree.ts:12]: We can later optimize with additional properties like `lastModified`, `updateFrequency`, and `priority`.

File: apps/frontpage/scripts/composeOtherSitemaps.ts [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+async function getRemoteSitemapContent(path: string) {
+  const response = await fetch(`https://storybook.js.org${path}`);
+  const content = await response.text();
+  return content;
+}
+
+const OTHER_SITEMAPS = {
+  addons: {
+    async getContent() {
```
Comment (kylegach) [inline apps/frontpage/scripts/composeOtherSitemaps.ts:13]: This approach appears to be ridiculously over-engineered. But it will allow to update these one-by-one to simply fetch a local sitemap file, once an app is in this monorepo.

File: package.json [hunk 3]
```diff
@@ -98,6 +98,7 @@
     "eslint-plugin-react-hooks": "4.6.0",
     "jest": "^29.4.1",
     "jest-environment-jsdom": "^29.4.1",
+    "next-sitemap": "^4.2.3",
```
Comment (cdedreuille) [inline package.json:101]: @kylegach Why did you need to use this package? I though Next.js managed sitemap on their own now. https://nextjs.org/docs/app/api-reference/file-conventions/metadata/sitemap#generating-a-sitemap-using-code-js-ts

File: libs/utils/src/lib/docs-versions.tsx [hunk 4]
```diff
@@ -2,31 +2,46 @@
...
   label: string;
+  /** Used for routes */
   id: string;
-  branch?: string;
-  tag?: string;
-  commit?: string;
+  /** Used in calculations */
+  number: number;
+  isPreRelease?: boolean;
 }
```
Comment (cdedreuille) [inline libs/utils/src/lib/docs-versions.tsx:25]: I'm a bit confused at what's happening here. In no circumstances we will update this array in the app so I'm wondering why adding more complexity to it. Looking at the type definition, I'm just a bit lost.

File: libs/utils/src/lib/docs-versions.tsx [hunk 5]
```diff
@@ -2,31 +2,46 @@
...
+type DocsVersionReferenceKind = 'branch' | 'commit' | 'tag';
+
+export type DocsVersion = BaseDocsVersion & ExactlyOneKeyWithType<DocsVersionReferenceKind, string>;
+
+/** Latest version must ALWAYS be first */
 export const docsVersions: DocsVersion[] = [
   {
     label: "Version Test 1",
     id: "8.0",
+    number: 8,
```
Comment (cdedreuille) [inline libs/utils/src/lib/docs-versions.tsx:37]: What is the difference between `number` and `id`? Why do you need the two?

File: apps/frontpage/scripts/getLocalDocs.ts [hunk 6]
```diff
@@ -2,9 +2,7 @@ import path from 'path';
 import fs from 'fs-extra';
 import fetch from 'node-fetch';
 import tar from 'tar';
-// TODO: Figure out why we can't use an absolute import here
-// import { DocsVersion, docsVersions } from "../docs-versions";
-import { DocsVersion, docsVersions } from './docs-versions';
+import { DocsVersion, docsVersions } from '@utils';
```
Comment (cdedreuille) [inline apps/frontpage/scripts/getLocalDocs.ts:5]: @kylegach This did work? On my side it did not. Not sure why.

File: package.json (hunk 3 above)
Comment (kylegach) [reply package.json:101]: I tried that approach and it didn't work in production. 😭 `next dev` & `next start` worked fine, local and Netlify builds did not. I tried for a couple hours. It seems to be a [not-uncommon problem](https://www.google.com/search?q=nextjs+%22sitemap.ts%22+production+-%22next-sitemap%22). You can actually see that your prior attempt using `sitemap.ts` never worked either: https://deploy-preview-21--storybook-web.netlify.app/sitemap.xml (linking to the previous deploy preview for posterity).

File: libs/utils/src/lib/docs-versions.tsx (hunk 5 above)
Comment (kylegach) [reply libs/utils/src/lib/docs-versions.tsx]: `number` is a number, used for calculations (e.g. `if (version < 8) { ... }`). `id` is a string, used for routes. I left comments in the type definition.

File: apps/frontpage/scripts/getLocalDocs.ts (hunk 6 above)
Comment (kylegach) [reply apps/frontpage/scripts/getLocalDocs.ts:5]: Yes, I got it to work. Here's the key, in `apps/frontpage/project.json`:

```diff
"fetch-docs": {
-  "command": "tsx apps/frontpage/scripts/getLocalDocs.ts"
+  "command": "tsx --tsconfig apps/frontpage/tsconfig.json apps/frontpage/scripts/getLocalDocs.ts"
},
```

This just helped me realize that I should add that to the other `tsx` usage, too.

File: libs/utils/src/lib/docs-versions.tsx (hunk 4 above)
Comment (kylegach) [reply libs/utils/src/lib/docs-versions.tsx]: Actually, another change made these benefits moot. I had to pull the actual `docsVersions` definition into a `/docs-version.js` file (which had to be CJS) at the project root, so that I could require it inside a CJS module for next-sitemap's config. Then I import that definition into this file, apply the type, and re-export it.

This complex typing was only to aid the maintenance of the `docsVersions` definition. But because of the CJS shenanigans, we cannot actually type that definition anymore, so there's no benefit.

So, long story, short... I removed all this complexity.
